### ISSUE TITLE
"index.js" references don't get resolved

### ISSUE THREAD

**Issue by smeijer** [davidmfoley/node-trucker#20]
Folder structure:

```
src
├── main.js
└── routes
    ├── account.js
    ├── blog.js
    └── index.js
```

main.js
```js
import routes from './routes';
```

routes/index.js
```js 
import account from './account';
import blogs from './blog';

export default { 
  account, 
  blogs, 
};
```

---

`routes/index.js` is referenced here by the `./routes` import in `main.js`. Still; `Trucker` marks `/routes/index.js` as being `unused`.

**Comment by davidmfoley** [comment]
Hmmm, not sure if I can repro this.

Here's how I am testing:

```bash
$ mkdir trucker-test
$ cd trucker-test
$ echo "import routes from './routes';" > main.js
$ mkdir routes
$ echo 'export default  { foo: () => console.log("what"); };' > routes/index.js
$ trucker -u
main.js
$ trucker -i routes/index.js
routes/index.js
 <--- main.js
$ trucker -i main.js
main.js
 ---> routes/index.js
```

**Comment by smeijer** [comment]
Could it be windows related?

```bash
mkdir trucker-test
cd trucker-test
echo "import routes from './routes';" > main.js
mkdir routes
echo 'export default  { foo: () => console.log("what"); };' > routes/index.js

trucker -u
# » main.js
# » routes\index.js
```

![image](https://user-images.githubusercontent.com/1196524/42754304-4ed6a6fe-88f4-11e8-861a-34118625f094.png)

**Comment by davidmfoley** [comment]
I think I fixed this in v0.7.6. Would you mind letting me know if you get a chance to test it? Thanks!

**Comment by smeijer** [comment]
Sorry, using `v0.7.7` but still having the same output.

```
trucker -u
# » main.js
# » routes\index.js
```

**Comment by davidmfoley** [comment]
@smeijer I finally got around to testing on windows and I believe the two underlying issues are now fixed: 

1. A couple of places didn't handle \ vs / for path.sep
2. File encodings are now detected and trucker *attempts* to decode based on the encoding. The test case above (on Win10 at least) created UTF-16 files and trucker was hardcoded to try to decode using UTF-8. 

Let me know if you get a chance to test it in your environment (0.8.2 or later). I tested running inside  powershell on Win10 (in VirtualBox).

Thanks
